### PR TITLE
[ErrorHandler] ErrorHandler - add filename and line number to error message

### PR DESCRIPTION
**Pull request by soleuu**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no
| Issues        | -
| License       | MIT

Filename and line number are missing from Logged Message

For debugging purpose we need filename and line number.

### CODE REVIEW

**Review by yceruto — APPROVED** [review]
LGTM as a bug fix, but it may also be considered a new feature.

**Review by derrabus — CHANGES_REQUESTED** [review]
Sorry, but that's a 👎🏻  from me. 😕 

This is certainly not a bugfix. I would not change the old LTS for that.

We can discuss this as a feature, but even there I would question if this change is really necessary. Along with the error message, we log an exception object with a full stack trace into the context of the log message. Rendering file and line into the message could be achieved with a monolog message formatter for instance, if someone really wanted that.

And if we decide that we always want to include file and line in our log message, I'm not happy with this implementation because it also changes the message of the logged exception to include that information. Given that the exception has dedicated properties for file and line, appending that information to the exception message is super redundant.

File: src/Symfony/Component/ErrorHandler/Tests/ErrorHandlerTest.php [hunk 1]
```diff
@@ -398,7 +398,7 @@ public function testHandleDeprecation()
             $this->assertArrayHasKey('exception', $context);
             $exception = $context['exception'];
             $this->assertInstanceOf(\ErrorException::class, $exception);
-            $this->assertSame('User Deprecated: Foo deprecation', $exception->getMessage());
+            $this->assertSame('User Deprecated: Foo deprecation at foo.php line 12', $exception->getMessage());
```
Comment (derrabus) [inline src/Symfony/Component/ErrorHandler/Tests/ErrorHandlerTest.php:401]: I can understand that we might want to see that information in our logs, but encoding it into the _exception_ message does not feel right.

Comment (smnandre) [reply src/Symfony/Component/ErrorHandler/Tests/ErrorHandlerTest.php:401]: I am indeed afraid adding file and line in the message could make deduplication or "grouping" harder.

**Comment by nicolas-grekas** [comment]
Hi, thanks for the PR but I agree with @derrabus: the thing you're trying to improve is a rendering issue. The message itself shouldn't contain pre-rendered metadata.

File: src/Symfony/Component/ErrorHandler/Tests/ErrorHandlerTest.php (hunk 1 above)
Comment (soleuu) [reply src/Symfony/Component/ErrorHandler/Tests/ErrorHandlerTest.php:401]: If there is a problem with exception Message then i can move the addition just before sending to the logger and let this test as it was before

**Comment by soleuu** [comment]
Hi,

@yceruto thank you.

@derrabus : I really consider this as a bug fix. ErrorHandler is just an override of what php do natively.
=> When the default php error_handler is in action, the error message is logged with a specific format.
"{time} {logLevel} : {Message} at {filename} in {line}"
=> The ErrorHandler class is just a re-implementation of what php do natively. So it must do at least the same thing. The current implementation does not show filename and line number.


@nicolas-grekas : This PR is just a way to resolve the problem I have (and I think many people have) but there are many way to resolve a problem. If you have a better solution, feel free to explain us how.

**Comment by derrabus** [comment]
> If you have a better solution, feel free to explain us how.

Quoting myself:

> Along with the error message, we log an exception object with a full stack trace into the context of the log message. Rendering file and line into the message could be achieved with a monolog message formatter for instance, if someone really wanted that.

**Comment by soleuu** [comment]
> > If you have a better solution, feel free to explain us how.
> 
> Quoting myself:
> 
> > Along with the error message, we log an exception object with a full stack trace into the context of the log message. Rendering file and line into the message could be achieved with a monolog message formatter for instance, if someone really wanted that.


I understand what you mean and thank you. 
But using monolog is like using a nuclear reactor when you just need a small battery.

Adding a new dependency to a project just because the core framework do not handle things correctly is just overkill.

A new dependency is a new software to keep updated, to upgrade (to migrate), ... with potential new security issues and so on.

I have made a new push taking note of what you said about changing Exception Message.


Do i need to make a new pull request ? Or is it possible to talk about it here.

Thank you for your time.

**Pull request closed without merging**